resources/macos_user.rb

```diff
@@ -25,6 +26,22 @@ def sysadminctl
     '/usr/sbin/sysadminctl'
   end
 
+  def full_name?
```

Comment:
This method is not a boolean, remove the `?`.

Reply:
What would be a good function name for a function returns whether "-fullName" is used or not? Also should we make the same change for the admin? routine since that also returns a string.

Reply:
We can put the logic into a single ternary statement:

```ruby
  def user_fullname
    property_is_set?(:fullname) ? "-fullName #{new_resource.fullname}" : ''
  end
```

And implement it in our execute block like this:

```ruby
  execute "add user #{new_resource.username}" do
    command [sysadminctl, '-addUser', new_resource.username, user_fullname, '-password', new_resource.password, admin_user?]
    not_if { ::File.exist?(user_home) && user_already_exists? }
  end
```

Where `user_already_exists?` is an abstraction of the guard that is used in the `:delete` action:

```ruby
  def user_already_exists?
    users = shell_out!('/usr/bin/dscl', '.', '-list', '/users').stdout
    return true if users.include?(new_resource.username)
  end
```